closed/src/java.base/share/native/libjncrypto/NativeCrypto.c

```diff
@@ -261,13 +265,38 @@ static void printErrors(void)
     fflush(stderr);
 }
 
+/*
+ * We use a 8 digit map (AABBCCDD) to represent the version of openssl.
+ * AA is the major version,
+ * BB is the minor version,
+ * CC is the revision, and DD for the letter that could be present in any version.
+ * For example, if an openssl version is in this scheme 1.1.1k.
+ * The ossl_ver should be 10101011 ( k is the 11th letter value for DD here... ).
```

Comment:
This comment needs to be updated: versions are expressed as up to 9 hex digits (three for DDD). For example, `1.1.1k` would map to `0x1010100b`.

Reply:
We used e.g. 10101011 for version 1.1.1k where AA is 10, BB is 10, CC is 10 and DD is 11. After changing to hexadecimal format (0x1_01_01_011L), A will be 1, BB will be 01, CC will be 01 and DDD will be 011, right?